src/main/scala/analysis/solvers/FixPointSolver.scala

```diff
@@ -204,25 +204,26 @@ trait WorklistFixpointSolverWithReachability[N, T, L <: Lattice[T]] extends Work
   }
 }
 
-/** A pushDown worklist-based fixpoint solvers. Pushes the results of the analysis one node down. This is used to have
-  * the results of the pred node in the current node. ie. NODE 1: R0 = 69551bv64 RESULT LATTICE = {} NODE 2: R0 =
-  * MemLoad[R0 + 54bv64] RESULT LATTICE = {R0 = 69551bv64} NODE 3: R1 = 0bv64 RESULT LATTICE = {R0 = TOP} ...
-  *
-  * @tparam N
-  *   type of the elements in the worklist.
-  *
-  * Better implementation of the same thing
-  * https://github.com/cs-au-dk/TIP/blob/master/src/tip/solvers/FixpointSolvers.scala#L311
-  */
-trait PushDownWorklistFixpointSolver[N, T, L <: Lattice[T]] extends MapLatticeSolver[N, T, L] with LinkedHashSetWorklist[N] with Dependencies[N]:
+/**
+ * Common trait for pushdown worklist solvers
+ *
+ * @tparam N
+ *   type of the elements in the worklist.
+ * @tparam T Type of the elements in the sublattice
+ * @tparam L Sublattice
+ */
+trait PushDownWorklistFixPointFunctions[N, T, L <: Lattice[T]] extends LinkedHashSetWorklist[N]:
```

Comment:
Why does this need to exist at all? It just seems to make the inheritance more convoluted without adding anything at present. Just because TIP has very convoluted inheritance doesn't mean we should copy that.